rdkit/Chem/Draw/__init__.py

```diff
@@ -573,6 +571,238 @@ def _MolsToGridImage(mols, molsPerRow=3, subImgSize=(200, 200), legends=None,
+            You do not need to do that for empty cells at the end of a row; 
+            this function will automatically pad rows so that all rows are the same length.
+            
+            This function is useful when each row has some meaning,
+            for example the generation in a mass spectrometry fragmentation tree--refer to 
+            example at https://en.wikipedia.org/wiki/Fragmentation_(mass_spectrometry).
+            If you want to display a set molecules where each row does not have any specific meaning,
+            use MolsToGridImage instead.
+
+            This function nests data structures one additional level beyond the analogous function MolsToGridImage
+            (in which the molecules and legends are non-nested lists, 
+            and the highlight parameters are two-deep nested lists) 
+
+      RETURNS:
+
+        A grid of molecular images in one of these formats:
+        
+        - useSVG=False and returnPNG=False (default): A PIL object for a PNG image file
+
+        - useSVG=False and returnPNG=True: A PNG file 
+
+        - useSVG=True: An SVG string
+
+      EXAMPLES:
+
+        from rdkit import Chem
+        from rdkit.Chem.Draw import MolsMatrixToGridImage, rdMolDraw2D
+        FCl = Chem.MolFromSmiles("FCl")
+        mols_matrix = [[FCl, FCl], [FCl, None, FCl]]
+
+        # Minimal example: Only mols_matrix is supplied,
+        # result will be a drawing containing (where each row contains molecules):
+        # F-Cl    F-Cl
+        # F-Cl             F-Cl
+        img = MolsMatrixToGridImage(mols_matrix)
+        img.save("MolsMatrixToGridImage_minimal.png")
+        # img is a PIL object for a PNG image file like:
+        # <PIL.PngImagePlugin.PngImageFile image mode=RGB size=600x200 at 0x1648CC390>
+        # Drawing will be saved as PNG file MolsMatrixToGridImage_minimal.png
+
+        # Exhaustive example: All parameters are supplied,
+        # result will be a drawing containing (where each row of molecules is followed by a row of legends):
+        # 1 F-Cl 0             1 F-Cl 0
+        # no highlighting       bond highlighted         
+        # 1 F-Cl 0                                1 F-Cl 0
+        # sodium highlighted                       chloride and bond highlighted
+        legends_matrix = [["no highlighting", "bond highlighted"], 
+        ["F highlighted", "", "Cl and bond highlighted"]]
+        highlightAtomLists_matrix = [[[],[]], [[0], None, [1]]]
+        highlightBondLists_matrix = [[[],[0]], [[], None, [0]]]
+
+        dopts = rdMolDraw2D.MolDrawOptions()
+        dopts.addAtomIndices = True
+
+        img_file = MolsMatrixToGridImage(mols_matrix=mols_matrix, subImgSize=(300, 400), 
+        legends_matrix=legends_matrix, highlightAtomLists_matrix=highlightAtomLists_matrix, 
+        highlightBondLists_matrix=highlightBondLists_matrix, useSVG=False, returnPNG=True, drawOptions=dopts)
+        img_file.save("MolsMatrixToGridImage_exhaustive.png")
```

Comment:
I believe that this is wrong. I think that you should get back a string (binary string, but still a string) when `returnPNG = True`.
You get the PIL object when `returnPNG = False`

Reply:
Maybe I'm missing something. The example you just shared has `returnPNG=False`. As I would expect, that returns a PIL object. 
My comment above is about when you do `returnPNG=True`; in that case you should get a string back, but the code I was commenting on was treating it as a PIL object.

Reply:
My mistake, I thought I was using the same options in my code as in the docstring but I must've changed them sometime along the line. I'll revisit.